### DISCUSSION TITLE
napi@2.14.0

### DISCUSSION THREAD

**Discussion by Brooooooklyn** [napi-rs/napi-rs#1793] [General]
## What's Changed
* throw `is not a valid identifier` when generating getters and setters by @yoogoc in https://github.com/napi-rs/napi-rs/pull/1701
* style: clippy fix by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1711
* feat(napi,sys): implement Symbol.for by @tom-sherman in https://github.com/napi-rs/napi-rs/pull/1721
* build: Docker arm64 platform by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1731
* test: aginst bun by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1730
* chore(napi): make napi_register_module_v1 pub by @MarkusJx in https://github.com/napi-rs/napi-rs/pull/1747
* feat: bump msrv to 1.65 by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1755
* chore(cli): update ci template by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1771
* test: reduce memory usage while testing aginst worker thread by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1769
* Integrate with emnapi by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1669
* fix(napi): potential double free issue by @kuzeyardabulut in https://github.com/napi-rs/napi-rs/pull/1679
* feat(napi): impl Default for Buffer by @ozgrakkurt in https://github.com/napi-rs/napi-rs/pull/1734
* fix(napi-derive): import `FromNapiValue` and `ToNapiValue` from `napi::bindgen_prelude` when generating an `enum` by @jzeuzs in https://github.com/napi-rs/napi-rs/pull/1713
* fix(napi): asan caught memory safety issue by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1776
* feat(napi): support async class factory by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1779
* feat(napi,sys): sync to napi9 by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1781
* feat: add experimental `Env::get_module_file_name` method by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1783
* test(napi): add tests for napi9 features by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1784
* style(napi): fix compile warning by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1785
* style(napi): as ptr => ptr::cast by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1786
* fix(napi): add tokio cleanup hook for more platforms by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1790
* fix(build): rerun if cli version changed by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1789
* feat(napi): pass the rest of async tests by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1792

## New Contributors
* @yoogoc made their first contribution in https://github.com/napi-rs/napi-rs/pull/1701
* @tom-sherman made their first contribution in https://github.com/napi-rs/napi-rs/pull/1721
* @kuzeyardabulut made their first contribution in https://github.com/napi-rs/napi-rs/pull/1679
* @RaphaelDarley made their first contribution in https://github.com/napi-rs/napi-rs/pull/1716
* @ozgrakkurt made their first contribution in https://github.com/napi-rs/napi-rs/pull/1734
* @jzeuzs made their first contribution in https://github.com/napi-rs/napi-rs/pull/1713

**Full Changelog**: https://github.com/napi-rs/napi-rs/compare/napi@2.13.3...napi@2.14.0

<hr /><em>This discussion was created from the release <a href='https://github.com/napi-rs/napi-rs/releases/tag/napi%402.14.0'>napi@2.14.0</a>.</em>

**Comment by belgattitude** [comment]
Reading

> Integrate with emnapi by @Brooooooklyn in https://github.com/napi-rs/napi-rs/pull/1669

does it mean support for wasm target ?

btw thanks for your work, I really appreciate.

**Reply by Brooooooklyn** [reply]
Yes, but there is also some follow-up work to be done to enable NAPI-RS to support out-of-the-box WASM support, like: https://github.com/napi-rs/napi-rs/issues/1794
